src/targeting/Product.mli

```diff
@@ -0,0 +1,5 @@
+(* This modules uses the ocaml type system to prove that the
+   Input_to_core_t.product is the same as
+   Semgrep_output_v1_t.product. *)
+
+val all : Semgrep_output_v1_t.product list
```

Comment:
you could maybe move this file in core/

Reply:
it's just not just about targeting.
You could even have 'type t = Semgrep_output_v1.product` in there,
so everywhere else we can use Product.t when we talk about the product, which is meant to become a core
data structure (hence the move to src/core/).